site/app/templates/notebook/Notebook.twig

```diff
@@ -262,6 +271,38 @@
 </div>
 
 <script>
+    {% if is_timed and allowed_minutes > 25 %}
+        //for displaying popup submit button
+        window.onload = function(){
+            sessionStorage.setItem("show_popup_submit", "yes");
+            setInterval(count, 1000);
+        };
+
+        let counter = 0;
+        function count(){
+            counter++;
+            //interval is 20 percent of overall time allowed or 10 minutes, whichever is greater
+            let interval = ((({{ allowed_minutes * 60 }}) / 5) > 600 ? ({{ allowed_minutes * 60 }}) / 5 : 600);
+            if (counter > interval) {
+                if (sessionStorage.getItem("show_popup_submit") === 'yes') {
+                    $(".notebook-submit-popup").show();
+                    counter = 0;
+                }
+            }
+        }
```

Comment:
This is not an accurate way to measure time. setInterval of 1000 ms looks like it will delay exactly 1 second but it is never really exact leading to your recorded time to go out of phase with the actual time. This also will not work if the page is refreshed as it resets counter back to 0. You are assuming that this script stays running the whole exam. Keep in mind the page is refreshed every time a submission is made.

Take a look at my gradeable timer PR or just the submission-page.js file to see how I made a more accurate timer.